### DISCUSSION TITLE
Support javax or jakarka api or both?

### DISCUSSION THREAD

**Discussion by chengfang** [jberet/jberet-jpa-repository#4] [Ideas]
Do you think this project, jberet-jpa-repository should support javax api, jakarta api, or both?

javax package name is used in Jakarta EE 8 and Java EE. jberet 1.x implements javax.batch api.

Jakarta EE 9 and 10 (10 is the current version, to be released soon) use jakarta api with jakarta prefix in package names. Jakarta batch 2.1 (the current version) has jakarta package name. WildFly version 27 (the current development version) aims to implement Jakarta EE 10, and so will support Jakarta EE api including jakarta batch 2.1.

Currently jberet-jpa-repository references javax.persistence api, probably other javax api. If we want to support jakarta api, we'll need to replace them with their jakarta counterparts.

If we want to support both, we should have 2 submodules: javax and jakarta, to produce jberet-job-repository and jberet-job-repository-jakarta artifacts. This can be done with a transformer easily. See https://github.com/wildfly/jboss-ejb-client/ . Or we can keep them in separate branches, which is cleaner in the long term.

**Comment by chengfang** [comment]
@amoscatelli WDYT?

**Comment by amoscatelli** [comment]
I think it's best to support jakarta only.
I was pretty sure I used only jakarta packages, if not it was a mistake during the migration to the new git repository.
My bad.

I'll try to fix this in an hour or two.

**Comment by amoscatelli** [comment]
@chengfang I made a pull request 
https://github.com/jberet/jberet-jpa-repository/pull/5

This should be ok,
please tell me what you think

**Comment by chengfang** [comment]
Cool, we'll aim for jakarta support only.
